### ISSUE TITLE
Support multiple configuration languages

### ISSUE THREAD

**Issue by pbnj** [domust/linker#20]
This is a neat project.

How do you feel about supporting multiple configuration languages besides HCL?

A library like [`viper`](https://github.com/spf13/viper) gives you support for `JSON, TOML, YAML, HCL, INI, envfile or Java properties formats`. I have used it before and it's used in many large projects as well.

**Comment by domust** [comment]
I was planning to add support for JSON anyway, in fact, the package itself already supports it, I just haven't added it to the binary yet.

I drew most of my inspiration for this project from https://terraform.io and https://suckless.org

I liked *terraform*'s declarative nature, builtin configuration formatting and resource graphs, which made it easy to understand what the tool was planning to do, but I didn't like people throwing away their statefiles when something went wrong. Because of this I'm excluding statefiles from this project.
Furthermore, I thought that if I used HCL as a configuration language, I would be able to tap into terraform libraries and bring some of those goodies here.

What I liked about *suckless* tools is their minimal nature, for example, their *surf* browser which doesn't even support tabs or bookmarks, in fact, none of their tools support tabs, but they have *tabbed* which can be easily combined with their other tools to add support for tabs. What I didn't like was the need to edit C files and compile them yourself, I would rather have a library for users to use instead of making the tool customizable by editing source or config files. Therefore, I would like to keep configuration languages to the bare minimum.

The book https://en.wikipedia.org/wiki/The_Art_of_Unix_Programming also had major influence.
I was thinking about adding support for taking input from stdout(or any reader for that matter), which would allow users to write their configuration in arbitrary format, then they could pipe it to some transformation tool and later pipe it to the linker. I've noticed that quite a few unix utils support JSON as and input, output or both. This would probably enable use cases I'm not able to think of yet.

I hope this clearly states my position.
